res/controllers/Stanton-DJC-4-scripts.js

```diff
@@ -0,0 +1,359 @@
+            }
+        },
+        unshift: function() {
+            this.inKey = "Move";
+        },
+        shift: function() {
+            this.inKey = "Scroll";
+        },
+    });
+
+    djc4.deck = [];
+    for (i = 0; i < 4; i++) {
+        djc4.deck[i] = new djc4.Deck(i + 1);
+        djc4.deck[i].setCurrentDeck("[Channel" + (i + 1) + "]");
+    }
+
+    djc4.effectUnit = [];
+    for (i = 0; i <= 3; i++) {
+        djc4.effectUnit[i] = new components.EffectUnit([i + 1]);
+        djc4.effectUnit[i].shiftOffset = 0x32;
+        djc4.effectUnit[i].shiftControl = true;
+        djc4.effectUnit[i].enableButtons[1].midi = [0x90 + i, 0x1F];
+        djc4.effectUnit[i].enableButtons[2].midi = [0x90 + i, 0x20];
+        djc4.effectUnit[i].enableButtons[3].midi = [0x90 + i, 0x21];
+        djc4.effectUnit[i].effectFocusButton.midi = [0x90 + i, 0x1D];
+        djc4.effectUnit[i].knobs[1].midi = [0xB0 + i, 0x09];
+        djc4.effectUnit[i].knobs[2].midi = [0xB0 + i, 0x0A];
+        djc4.effectUnit[i].knobs[3].midi = [0xB0 + i, 0x0B];
+        djc4.effectUnit[i].dryWetKnob.midi = [0xB0 + i, 0x08];
+        djc4.effectUnit[i].dryWetKnob.input = function(channel, control, value) {
+            if (value === 0x41) {
+                // 0.05 is an example. Adjust that value to whatever works well for your controller.
+                this.inSetParameter(this.inGetParameter() + 0.05);
+            } else if (value === 0x3F) {
+                this.inSetParameter(this.inGetParameter() - 0.05);
+            }
+        };
+        djc4.effectUnit[i].init();
+    }
+
+    // === VU Meter ===
+    djc4.vuMeter = new components.Component({
+        midi: [0xB0, 0x03],
+        group: "[Master]",
+        outKey: "VuMeterL",
+        output: function(value, group) {
+            // The red LEDs light up with MIDI values greater than 0x60.
+            // The Red LEDs should only be illuminated if the track is clipping.
+            if (engine.getValue(group, "PeakIndicator") === 1) {
+                value = 0x60;
+            } else {
+                value = Math.round(value * 0x54);
+            }
+            this.send(value);
+        },
+    });
+
+    djc4.vuMeter = new components.Component({
+        midi: [0xB0, 0x04],
+        group: "[Master]",
+        outKey: "VuMeterR",
+        output: function(value, group) {
+            // The red LEDs light up with MIDI values greater than 0x60.
+            // The Red LEDs should only be illuminated if the track is clipping.
+            if (engine.getValue(group, "PeakIndicator") === 1) {
+                value = 0x60;
+            } else {
+                value = Math.round(value * 0x54);
+            }
+            this.send(value);
+        },
+    });
+};
+
+// Called when the MIDI device is closed
+djc4.shutdown = function() {
+    // Put all LEDs to default state.
+    djc4.allLed2Default();
+};
+
+djc4.Deck = function(deckNumbers) {
```

Comment:
The scripts expects a single integer for the deck so I'd rename this everywhere to `deckNumber` instead of `deckNumbers`